### PR TITLE
Monica's Final Report

### PR DESCRIPTION
**Pull request by monicayao**
🌟

### CODE REVIEW

File: _posts/blog/2019-08-14-Monica's-Final.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+---
+layout: post
+title: Google Summer of Code - Monica's Final Report
+categories: blog
+excerpt: As summer comes to an end, so does GSoC. Monica reflects about her time here.
```
Comment (KirstieJane) [inline _posts/blog/2019-08-14-Monica's-Final.md]: Syncing across your and @wingedRuslan's post.

```suggestion
excerpt: Final report for work on the tedana project during GSoC 2019
```

File: _posts/blog/2019-08-14-Monica's-Final.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+---
+layout: post
+title: Google Summer of Code - Monica's Final Report
```
Comment (KirstieJane) [inline _posts/blog/2019-08-14-Monica's-Final.md]: Just changing to sync up your and @wingedRuslan's posts.

```suggestion
title: Monica's GSoC Final Report
```

File: _posts/blog/2019-08-14-Monica's-Final.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+First, let me bring back an excerpt from my introduction:
+
+> From this summer, I wish to achieve a couple of things: 
+> 1. Learn about the awesome and neat science behind multi-echo fMRI
+> 2. How this library could be advanced in the future, should there be a new publication about denoising ME fMRI data
+> 3. Adding lots and lots of tests to the library so that contributors in the future could extend the library without worries
+> 4. Familiarizing myself with open source, and continue adding onto libraries as a regular contributor!
+
+This is what I wanted to do this summer! It's time to reflect.
```
Comment (KirstieJane) [inline _posts/blog/2019-08-14-Monica's-Final.md]: Please can you put all sentences on separate lines? Thank you!

(I haven't marked this all the way down - have a read through and find those ones that need to change.)

```suggestion
This is what I wanted to do this summer!
It's time to reflect.
```

File: _posts/blog/2019-08-14-Monica's-Final.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+### What I wanted to do
+
+I wanted to add a unit test to all the different functions in the tedana library, making it future proof for those who contribute to it in the future.
```
Comment (KirstieJane) [inline _posts/blog/2019-08-14-Monica's-Final.md]:
```suggestion
I wanted to add a unit test to all the different functions in the `tedana` library, making it future proof for those who contribute to it in the future. 
```

File: _posts/blog/2019-08-14-Monica's-Final.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+One big reason being that many who utilize `tedana` in publications are neuroscientists who do not regularly code, but the `tedana` community members want to welcome them to!
+
+This project acheives that goal because it would be a lot more welcoming to have simple unit test functions that not only show them what the functions are doing, but also catch any potential bugs or silly mistakes they could make.
+
+### What I did 
+
+In the end, I added a "smoke test" to all the functions. Smoke tests simply test the different I've recently started on adding a unit test for all the functions.
+
+Links of some of the pull requests and issues! 
+https://github.com/ME-ICA/tedana/pull/313
```
Comment (KirstieJane) [inline _posts/blog/2019-08-14-Monica's-Final.md]: Format as bullet points and add titles to list.

Can you format it in the same way as @wingedRuslan (note that you both have to change the list formatting so have a chat about whatever looks best!) https://github.com/WhitakerLab/whitakerlab.github.io/pull/74/files#diff-1e701786cbd66a31d09d8375a3b22e05R45 - line 45

File: _posts/blog/2019-08-14-Monica's-Final.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+https://github.com/ME-ICA/tedana/pull/314
+https://github.com/ME-ICA/tedana/pull/366
+https://github.com/ME-ICA/tedana/pull/367
+https://github.com/ME-ICA/tedana/issues/290
+https://github.com/ME-ICA/tedana/issues/335
+https://github.com/ME-ICA/tedana/issues/376
+
+I don't have a comparison of the before/after of test coverage but a recent screenshot from a recent code coverage graph!
```
Comment (KirstieJane) [inline _posts/blog/2019-08-14-Monica's-Final.md]: Can you check with Josh to see if it's possible that codecov keeps a history? Would be fun to see the difference...

File: _posts/blog/2019-08-14-Monica's-Final.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+Did not acheive all my hope and dreams - adding a unit test to every single function.
+
+The reason was simple, lots in my life happened this summer. Many different times, I wanted to flat out quit GSoC and focus on managing the stress that came from life's challenges. 
+Nonetheless, I'm extremely grateful to Kirstie and the `tedana` members for allowing me to stay onboard for the project, and guiding me along the way!
+
+
+### What was hard
+
+Getting used to the workflow. Open source is extremely independent but also extremely dependent on the members of the community. A big problem I had was wanting to work by myself to understand the library and overcome the different roadblocks I encountered. Many of those would have been solved by the members, yet I was stubborn.
```
Comment (KirstieJane) [inline _posts/blog/2019-08-14-Monica's-Final.md]: This is a great paragraph - I think it's true that this was a big barrier this summer.

Can you add a sentence/paragraph about what you'd do difference, and/or advice to future GSOC students who might feel the same way?

File: _posts/blog/2019-08-14-Monica's-Final.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+
+### What was hard
+
+Getting used to the workflow. Open source is extremely independent but also extremely dependent on the members of the community. A big problem I had was wanting to work by myself to understand the library and overcome the different roadblocks I encountered. Many of those would have been solved by the members, yet I was stubborn. 
+
+### What I learned
+
+Testing!!! There are so many different ways to test functions. Just in the tedana library, there are break tests, smoke tests, unit tests, integration tests. Quality assurance isn't as easy as it seems. 
+In addition, I learned that the use of github isn't just version control, it is for you to make a pull request from your code, so others can review it. So what if it is rejected?!
```
Comment (KirstieJane) [inline _posts/blog/2019-08-14-Monica's-Final.md]: I think the point about having your PR rejected could be expanded - at the moment it looks a little throw away and would make me feel _more_ anxious_ about opening a PR!!

File: _posts/blog/2019-08-14-Monica's-Final.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+### What I’m going to do next (not related to the project!)
+As cheesy as it sounds, this summer made me realize that a lot of what you plan do not come to actualization. So many different circumstances appeared, and I have learned to deal with some stressful real-life experiences. The next step for me is to focus on myself. I don’t think I’m at a good stage for more work, and I want to take a break from school (for a semester or a year). It may be too late for the fall semester, but I think that I will take the spring off to go back to China and spend a lot more time with family and getting to know my culture!
```
Comment (KirstieJane) [inline _posts/blog/2019-08-14-Monica's-Final.md]: 💖 I think that sounds like a great plan. Thank you for sharing.

File: _posts/blog/2019-08-14-Monica's-Final.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+### What was hard
+
+Getting used to the workflow. Open source is extremely independent but also extremely dependent on the members of the community. A big problem I had was wanting to work by myself to understand the library and overcome the different roadblocks I encountered. Many of those would have been solved by the members, yet I was stubborn. 
+
+### What I learned
+
+Testing!!! There are so many different ways to test functions. Just in the tedana library, there are break tests, smoke tests, unit tests, integration tests. Quality assurance isn't as easy as it seems. 
+In addition, I learned that the use of github isn't just version control, it is for you to make a pull request from your code, so others can review it. So what if it is rejected?!
+
+### What I’m going to do next (not related to the project!)
```
Comment (KirstieJane) [inline _posts/blog/2019-08-14-Monica's-Final.md]: This heading is going to format weirdly long I think

```suggestion
### What I’m going to do next
```

File: _posts/blog/2019-08-14-Monica's-Final.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+> From this summer, I wish to achieve a couple of things: 
+> 1. Learn about the awesome and neat science behind multi-echo fMRI
+> 2. How this library could be advanced in the future, should there be a new publication about denoising ME fMRI data
+> 3. Adding lots and lots of tests to the library so that contributors in the future could extend the library without worries
+> 4. Familiarizing myself with open source, and continue adding onto libraries as a regular contributor!
+
+This is what I wanted to do this summer! It's time to reflect.
+
+### What I wanted to do
```
Comment (KirstieJane) [inline _posts/blog/2019-08-14-Monica's-Final.md]: I think this section needs a little bit more about what `tedana` is and why testing is useful.

Just a bit more background for readers who don't already know.

This is where you can add a screenshot of "tests passing" - the output from pytest in the command line.

**Review by KirstieJane — COMMENTED** [review]
This is great @monicayao!! Well done!

A few comments and just a little more to bulk out.

Please could you thank anyone in the tedana team who have helped you - I'd love for Josh and Taylor to get a shout out if you think that's fair.

Also - I don't think you can have a `'` in your file name - would you mind changing it to `2019-08-20-GSOC2019-Monica-Final-Report.md`? ~If you can also change the date to next Tuesday that makes most sense I think.~ Keep the date as it is and we can change it before we post. If you put a date in the future it won't show up when you build it locally!

[EDIT: you can have a `'` so I'll change this to just, I don't like having the `'` in the filename 😆]

File: _posts/blog/2019-08-14-Monica's-Final.md (hunk 7 above)
Comment (KirstieJane) [reply _posts/blog/2019-08-14-Monica's-Final.md]: I think you also had a lot of difficulty at the start understanding all the complexities of MRI physics and fMRI analyses!! Please do add that in - its absolutely true that `tedana` is a tough project to get started with!

**Review by KirstieJane — APPROVED** [review]
This looks awesome @monicayao!!

Can you pull from master and see if the CI passes now that I *think* I've fixed travis??

Its all approved from my side!

**Comment by monicayao** [comment]
@KirstieJane Done! It passed Travis CI 😄

**Pull request merged**